tests/unit/mesh/data/test_gridfromoptions.cxx

```diff
@@ -11,240 +11,278 @@
 #include <string>
 #include <vector>
 
-/// Global mesh
-namespace bout {
-namespace globals {
-extern Mesh* mesh;
-} // namespace globals
-} // namespace bout
-
 // The unit tests use the global mesh
 using namespace bout::globals;
 
 class GridFromOptionsTest : public FakeMeshFixture {
 public:
-  GridFromOptionsTest() : FakeMeshFixture(), options(), griddata(&options) {
+  GridFromOptionsTest() : FakeMeshFixture(), options(), griddata(nullptr) {
+
+    mesh_from_options.StaggerGrids = true;
+    mesh_from_options.xstart = 2;
+    mesh_from_options.xend = nx - 3;
+    mesh_from_options.ystart = 2;
+    mesh_from_options.yend = ny - 3;
+
+    mesh_from_options.createDefaultRegions();
+
     // We need a parallel transform as FieldFactory::create3D wants to
     // un-field-align the result
-    mesh->setParallelTransform(
-        bout::utils::make_unique<ParallelTransformIdentity>(*mesh));
+    mesh_from_options.setParallelTransform(
+        bout::utils::make_unique<ParallelTransformIdentity>(mesh_from_options));
+
+    griddata = new GridFromOptions(&options);
+    mesh_from_options.setGridDataSource(griddata);
 
     output_info.disable();
+    output_progress.disable();
     output_warn.disable();
     options["f"] = expected_string;
+
+    // modify mesh section in global options
+    options["dx"] = "1.";
+    options["dy"] = "1.";
+    options["g11"] = expected_string + " + 5.";
+    options["g22"] = expected_string + " + 4.";
+    options["g33"] = expected_string + " + 3.";
+    options["g12"] = expected_string + " + 2.";
+    options["g13"] = expected_string + " + 1.";
+    options["g23"] = expected_string;
+
+    mesh_from_options.getCoordinates();
+
+    expected_2d = {makeField<Field2D>(
+        [](Field2D::ind_type& index) {
+          return index.x() + (TWOPI * index.y()) + (TWOPI * index.z() / nz) + 3;
+        },
+        &mesh_from_options)};
+
+    expected_3d = {makeField<Field3D>(
+        [](Field3D::ind_type& index) {
+          return index.x() + (TWOPI * index.y()) + (TWOPI * index.z() / nz) + 3;
+        },
+        &mesh_from_options)};
   }
 
   ~GridFromOptionsTest() {
     Options::cleanup();
     output_info.enable();
+    output_progress.enable();
     output_warn.enable();
+    // note GridFromOptions* griddata will be deleted by the ~Mesh() destructor
```

Comment:
Ah, I was wondering why you made `griddata` a pointer -- I bet that was a fun discovery!

One day, we'll make `Mesh` store a `unique_ptr` rather than a raw pointer